src/main/java/frc/robot/leds/LedStrip.java

```diff
@@ -0,0 +1,192 @@
+        mode = Mode.SOLID;
+    }
+
+    /**
+     * Switch mode to blink.
+     * (Should use setBlinkTime before this)
+     */
+    public void blink() {
+        mode = Mode.BLINK;
+    }
+
+    /**
+     * Switch mode to fade.
+     * (Should use setFadeDuration before this)
+     */
+    public void fade() {
+        mode = Mode.FADE;
+    }
+
+    /**
+     * Switch mode to rainbow.
+     */
+    public void rainbow() {
+        mode = Mode.RAINBOW;
+    }
+
+    /**
+     * Sets the led blink time.
+     *
+     * @param blinkTime Blink time in seconds
+     */
+    public void setBlinkTime(double blinkTime) {
+        this.blinkTime = blinkTime;
+    }
+
+    /**
+     * Sets the duration of the fade effect.
+     *
+     * @param duration Fade duration in seconds.
+     */
+    public void setFadeDuration(double duration) {
+        fadeDuration = duration;
+    }
+
+    /**
+     * Sets the led strip to a solid color
+     *
+     * @param color Color to set the strip to.
+     */
+    private void setSolidColor(Color color) {
+        var appliedColor = new edu.wpi.first.wpilibj.util.Color(color.getRed(), color.getGreen(), color.getBlue());
+        for (int i = 0; i < ledBuffer.getLength(); i++) {
+            ledBuffer.setLED(i, appliedColor);
+        }
+        ledStrip.setData(ledBuffer);
+    }
+
+    /**
+     * Interpolates between two colors.
+     *
+     * @param initial Initial color.
+     * @param goal    Final Color.
+     * @return
+     */
+    private Translation3d colorInterpolation(Color initial, Color goal) {
+        var initialHSB = Color.RGBtoHSB(initial.getRed(), initial.getGreen(), initial.getBlue(), new float[3]);
+        var goalHSB = Color.RGBtoHSB(goal.getRed(), goal.getGreen(), goal.getBlue(), new float[3]);
+
+        Translation3d initialPoint = new Translation3d(initialHSB[0], initialHSB[1], initialHSB[2]);
+        Translation3d goalPoint = new Translation3d(goalHSB[0], goalHSB[1], goalHSB[2]);
+
+        double d = initialPoint.getDistance(goalPoint) / fadeDuration;
+        double t = d / (initialPoint.minus(goalPoint)).getNorm();
+
+        solution = initialPoint.interpolate(goalPoint, t);
+        return solution;
+    }
+
+
+    /**
+     * Sets the led strip to a rainbow pattern.
+     */
+    private void setRainbow() {
+        for (int i = 0; i < ledBuffer.getLength(); i++) {
+            ledBuffer.setHSV(i, rainbowHue, 255, 180);
+            rainbowHue += (180 / ledBuffer.getLength());
+            ledStrip.setData(ledBuffer);
+            rainbowHue %= 180;
+        }
+    }
+
+    @Override
+    public void periodic() {
+        if (mode == Mode.SOLID) {
+            setSolidColor(currentColor);
+        }
+        if (mode == Mode.BLINK) {
+            if (currentColor == primary) currentColor = secondary;
+            else currentColor = primary;
+
+            if (timer.hasElapsed(blinkTime)) {
+                setSolidColor(currentColor);
+            }
+            timer.reset();
```

Comment:
It's better to use [`Timer.advanceIfElapsed()`](https://github.wpilib.org/allwpilib/docs/release/java/edu/wpi/first/wpilibj/Timer.html#advanceIfElapsed(double)), it's more accurate. [Here](https://github.com/Galaxia5987/common/pull/2/#discussion_r1459466750)'s an explanation if the API reference isn't enough.

Reply:
You shouldn't reset the timer now, also, you don't check `timer.advanceIfElapsed(blinkTime)` return value, so I'm not sure what you're trying to check here. I recommend looking at that method's implementation/